state/state.go

```diff
@@ -306,6 +308,32 @@ func (s *State) CacheAndUnpack(b *ui.Task, p *manifest.Package) error {
 	return nil
 }
 
+// This is needed as if you run CacheAndUnpack on x86_64 mac for a manifest
+// with an entry for arm64 mac packages it does on extract that as
+// isExtracted returns true.
+// This method will only cache the values and try to get a Digest
+func (s *State) CacheAndDontUnpack(b *ui.Task, p *manifest.Package) (string, error) {
+	actualDigest := ""
+	if !s.isCached(p) {
+		mirrors := append(p.Mirrors, s.generateMirrors(p.Source)...)
+		_, _, ch, err := s.cache.Download(b, p.SHA256, p.Source, mirrors...)
+		if err != nil {
+			return "", errors.WithStack(err)
+		}
+		actualDigest = ch
+	} else {
+		path := s.cache.Path(p.SHA256, p.Source)
+		data, err := os.ReadFile(path)
+		if err != nil {
+			return "", errors.WithStack(err)
+		}
```

Comment:
Would `s.cache.Open` be appropriate here?

Reply:
Open is going to try and download and also do a stat. At this point I have already checked that it's cached so I just want to read -> hash and done.